cmd/upgrade.go

```diff
@@ -0,0 +1,145 @@
+package cmd
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/spf13/cobra"
+)
+
+var upgradeCmd = &cobra.Command{
+	Use:           "upgrade",
+	Short:         "Upgrade osdctl",
+	Long:          "Fetch latest osdctl from GitHub and replace the running binary",
+	RunE:          upgrade,
+	SilenceErrors: true,
+}
+
+func upgrade(cmd *cobra.Command, args []string) error {
+	// rootName ensures that the upgrade will fail if we ever decide to rename osdctl
+	// between releases :-)
+	rootName := cmd.Root().Name()
+
+	latest, err := getLatestVersion()
+	if err != nil {
+		return err
+	}
+	latestWithoutPrefix := strings.TrimPrefix(latest, "v")
+	// check if an upgrade is necessary
+	currentSemVer := semver.New(Version)
```

Comment:
Just curious. where does Version come from?

Reply:
Version is one of the variables we inject during the build process via Go build flags: https://github.com/openshift/osdctl/pull/196/files/ea42f9e990dbfa2857709584806749f22758536c#diff-7cb26babce207ed9c17e9dd3ead0b66d204da3cf4d11649412c197defbb3029cL18

Reply:
With Go 18 this kind of information is automatically embedded in `debug.GitVersion`, IIRC. This might be a nice task for the future :) Would make our build pipeline a little bit simpler

Reply:
@fahlmant this line
https://github.com/openshift/osdctl/blob/master/.goreleaser.yml#L18
Also @shibumi , can you add a comment here and there regarding this voodoo?

Reply:
Yes, a comment would be much appreciated :)

Reply:
@fahlmant  done